### PR TITLE
Setting to enable new experimental channel sidebar

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
This PR adds a new server config item that enables users to enable or disable the new experimental channel sidebar

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-21296

#### Related PRs
Webapp: https://github.com/mattermost/mattermost-webapp/pull/4999

### CODE REVIEW

File: report.xml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (cpanato) [inline report.xml]: do you need this file?

Comment (devinbinnie) [reply report.xml]: No, i'm not sure why it ended up in there...I'll remove it

**Comment by devinbinnie** [comment]
@cpanato @gsagula can you guys prioritize the review on this for today? we're planning on merging this and the webapp today.

**Comment by devinbinnie** [comment]
Thanks for the quick review guys.
Aside, do any of you know why this test would be failing:
```
--- FAIL: TestTermsOfServiceStore (0.01s)
    --- PASS: TestTermsOfServiceStore/LocalCache+MySQL (0.00s)
        --- PASS: TestTermsOfServiceStore/LocalCache+MySQL/TestSaveTermsOfService (0.00s)
        --- PASS: TestTermsOfServiceStore/LocalCache+MySQL/TestGetLatestTermsOfService (0.00s)
        --- PASS: TestTermsOfServiceStore/LocalCache+MySQL/TestGetTermsOfService (0.00s)
    --- FAIL: TestTermsOfServiceStore/LocalCache+PostgreSQL (0.00s)
        --- PASS: TestTermsOfServiceStore/LocalCache+PostgreSQL/TestSaveTermsOfService (0.00s)
        --- FAIL: TestTermsOfServiceStore/LocalCache+PostgreSQL/TestGetLatestTermsOfService (0.00s)
            terms_of_service_store.go:53: 
                	Error Trace:	terms_of_service_store.go:53
                	            				terms_of_service_store.go:17
                	Error:      	Not equal: 
                	            	expected: "7p5kieo3xp85iybgizsk5i7mqo"
                	            	actual  : "y1s7nrjn7in7jrbeaabgx9p7ha"
                	            	
                	            	Diff:
                	            	--- Expected
                	            	+++ Actual
                	            	@@ -1 +1 @@
                	            	-7p5kieo3xp85iybgizsk5i7mqo
                	            	+y1s7nrjn7in7jrbeaabgx9p7ha
                	Test:       	TestTermsOfServiceStore/LocalCache+PostgreSQL/TestGetLatestTermsOfService
        --- PASS: TestTermsOfServiceStore/LocalCache+PostgreSQL/TestGetTermsOfService (0.00s)
```

I didn't touch any files related to it AFAIK.

**Comment by devinbinnie** [comment]
@cpanato @gsagula Any ideas on this guys? https://github.com/mattermost/mattermost-server/pull/14036#issuecomment-598765456

**Comment by agnivade** [comment]
@devinbinnie - That's a flaky test: https://mattermost.atlassian.net/browse/MM-23063.

**Comment by devinbinnie** [comment]
> @devinbinnie - That's a flaky test: https://mattermost.atlassian.net/browse/MM-23063.

Thanks :) was wondering why it just went away lol.

**Pull request merged**